cmake/compiler_support.cmake

```diff
@@ -31,22 +31,65 @@ if (NOT PERL_FOUND )
    message(FATAL_ERROR "Perl not found")
 endif()
 
+
 set(MASM2GAS_PATH ${OMR_ROOT}/tools/compiler/scripts/masm2gas.pl CACHE INTERNAL "MASM2GAS PATH")
 
 include(${OMR_ROOT}/cmake/AddPrefix.cmake) 
 
-# Platform setup code!
-# TODO: THis needs to be abstracted for cross platform builds
+# Fetch the OMR view of the system.
+include(OmrDetectSystemInformation)
+
+macro(tr_detect_system_information)
+
+	macro(jit_not_ready)
+		message(FATAL "JIT isn't ready to build with CMake on this platform")
+	endmacro()
+
+
+	omr_detect_system_information()
+
+
+	set(TR_COMPILE_DEFINITIONS "")
+
+
+	# Platform setup code! 
+	# TODOs:
+	#  - Support more platforms, and, separate host and target arch. 
+	#  - Once we support all the platforms OMR supports with CMake, this can be
+	#    largely integrated into OmrDetectSystemInformation.cmake. 
+	if(OMR_ARCH_X86)
+		set(TR_HOST_ARCH    x     CACHE INTERNAL "The architecture directory used for the compiler code (x, p, or z)")
+		list(APPEND TR_COMPILE_DEFINITIONS TR_HOST_X86 TR_TARGET_X86)
+		if(OMR_ENV_DATA64)
+			set(TR_HOST_SUBARCH amd64 CACHE INTERNAL "The subarchitecture directory used for the compiler code. May be empty (i386 or amd64)")
+			set(TR_HOST_BITS    64    CACHE INTERNAL  "Bitness of the target architecture")
+			set(CMAKE_ASM-ATT_FLAGS "--64 --defsym TR_HOST_X86=1 --defsym TR_HOST_64BIT=1 --defsym BITVECTOR_64BIT=1 --defsym LINUX=1 --defsym TR_TARGET_X86=1 --defsym TR_TARGET_64BIT=1" CACHE INTERNAL "ASM FLags")
```

Comment:
This is something we need to look into further as we decide how we are going to handle ASM files.  These options will break other components using the assembler.  I do not see the assembler being enabled in CMake as a language yet for the compiler so this set line is not being used as far as I can tell. This is not new code but I am just pointing out as a future consideration before I forget :)